### PR TITLE
Cc/fe updates

### LINKED ISSUE #63: Add a revoke access button
To revoke a doctor's access to files

### LINKED ISSUE #64: Show doctor's name rather than eth address

### LINKED ISSUE #65: Implement private routes and auth restrictions
This is a large issue, and will involve some user flow decisions. However, it's important.

### LINKED ISSUE #66: Add spinner on sign up
Sign up takes a bit, add a spinner

### LINKED ISSUE #67: Handle web3 query errors
Currently console logs, but we need to handle errors in order to display them in the UI

### LINKED ISSUE #68: Add ability ti upload files
currently we have a text box

### PR DESCRIPTION
**Pull request by hack-heart**
closes #68, closes #65, closes #64, closes #63, closes #67

Updates to frontend

### CODE REVIEW

**Review by Scott-Crawshaw — APPROVED** [review]
Looks good!

File: src/common/CustomSpinner.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
+// inspired by https://davidwalsh.name/css-cube
```
Comment (SylvesterElormCoch) [inline src/common/CustomSpinner.js:1]: Ouu Fancyy :) I lovee

File: src/hooks/useAuth.js [hunk 2]
```diff
@@ -1,44 +0,0 @@
-// how to know when a user is a valid user: their wallet address
```
Comment (SylvesterElormCoch) [inline src/hooks/useAuth.js:1]: Is this completely useless?

File: src/common/LogoutButton.js [hunk 3]
```diff
@@ -1,14 +1,17 @@
...
-  </Button>
-);
+// It isn't possible to programmatically disconnect from a wallet. The user has to do it themselves
+// Add this information to walkthrough
+const LogoutButton = (props) => {
+  return (
+    <Button
+      aria-label="display more actions"
+      endIcon={<LogoutIcon />}
+    > DISCONNECT
```
Comment (SylvesterElormCoch) [inline src/common/LogoutButton.js:12]: Thinking this is could be a dropdown with options to toggle view, and also Disconnect - working on this but good to keep in PR for now

**Review by SylvesterElormCoch — APPROVED** [review]

File: src/admin/AdminSinglePatient.js [hunk 4]
```diff
@@ -213,34 +239,21 @@ const AdminSinglePatient = (props) => {
...
+                        label="File type"
+                        onChange={(event) => { setFileType(event.target.value); }}
+                      >
+                        <MenuItem value="Medical history">Medical history</MenuItem>
+                        <MenuItem value="Blood test">Blood test</MenuItem>
+                        <MenuItem value="Procedure">Procedure</MenuItem>
+                      </Select>
+                    </FormControl>
+                    <FormControl fullWidth size="small" margin="normal">
+                      <Input type="file" name="file" onChange={onFileUpload} />
```
[the remark this answers is not in the document]
Comment (hack-heart) [reply src/admin/AdminSinglePatient.js:255]: Gotcha. Thanks for the example!

**Pull request merged**